mne/time_frequency/tfr.py

```diff
@@ -1107,9 +1162,246 @@ def plot(self, picks, baseline=None, mode='mean', tmin=None, tmax=None,
+            to be used for averaging (seconds, Hertz). If a tuple is
+            provided, use the same window span for all timefreqs points
+            defined in `timefreqs`.For example, (0.01, 2) would translate
+            into a window that starts 5 ms before and ends 5 ms after a
+            given time point, as well as spanning 1 Hz before and after the
+            given frequency point.
+            If an iterable of tuples of the same length as `timefreqs` is
+            passed, then each tuple applies to its corresponding timefreqs
+            point. Defaults to None, which means no averaging.
+        topomap_args : None | dict
+            A dict of `kwargs` that are forwarded to `evoked.plot_topomap`
+            to style the topomaps. `axes` and `show` are ignored. If `times`
+            is not in this dict, automatic peak detection is used. Beyond that,
+            if ``None``, no customizable arguments will be passed.
+            Defaults to ``None``.
+
+        Returns
+        -------
+        fig : matplotlib.figure.Figure
+            The figure containing the topography.
+        """  # noqa: E501
+        from ..viz.topomap import _set_contour_locator
+        from ..viz.utils import _setup_vmin_vmax
+        import matplotlib.pyplot as plt
+
+        if topomap_args is None:
+            topomap_args = dict()
+
+        cmap = _setup_cmap(cmap)
+
+        fig = plt.figure()
+        tf_ax = fig.add_subplot(212)
+
+        _, tf_data, tf_times, tf_freqs = \
+                             self._plot(picks=picks, exclude=exclude, baseline=baseline,
+                             mode=mode,tmin=tmin, tmax=tmax, fmin=fmin, fmax=fmax,
+                             vmin=vmin, vmax=vmax, cmap=cmap, dB=dB,
+                             colorbar=False, show=show, title=title,
+                             axes=tf_ax, layout=layout, yscale=yscale,
+                             verbose=verbose, aggregate=aggregate)
+
+        vmin, vmax = _setup_vmin_vmax(tf_data, vmin, vmax)
+
+        if timefreqs is None:
+            # find the maximum peak
+            from scipy.signal import argrelmax
+            order = tf_data.shape[2] // 30
+            if order < 1:
+                order = 1
+            peaks_idx = argrelmax(tf_data, order=order, axis=2)
+            if peaks_idx[0].size == 0:
+                _,p_t,p_f = np.unravel_index(tf_data.argmax(), tf_data.shape)
+                timefreqs = [(tf_times[p_t], tf_freqs[p_f])]
+            else:
+                peaks = [tf_data[0,f,t] for f,t in zip(peaks_idx[1], peaks_idx[2])]
+                peakmax_idx = np.argmax(peaks)
+                peakmax_time = tf_times[peaks_idx[2][peakmax_idx]]
+                peakmax_freq = tf_freqs[peaks_idx[1][peakmax_idx]]
+
+                timefreqs = [(peakmax_time, peakmax_freq)]
+
+        # prepare axes for topomap
+        # slightly convoluted due to colorbar placement and for vertical alignment
```

Comment:
similarly - does it make sense to create a helper to use by both plot_joint functions?